packages/core/src/modules/connections/DidExchangeProtocol.ts

```diff
@@ -61,28 +62,38 @@ export class DidExchangeProtocol {
   }
 
   public async createRequest(
-    connectionRecord: ConnectionRecord,
+    outOfBandRecord: OutOfBandRecord,
     params: DidExchangeRequestParams
-  ): Promise<DidExchangeRequestMessage> {
-    this.logger.debug(`Create message ${DidExchangeRequestMessage.type} start`, connectionRecord)
+  ): Promise<{ message: DidExchangeRequestMessage; connectionRecord: ConnectionRecord }> {
+    this.logger.debug(`Create message ${DidExchangeRequestMessage.type} start`, { outOfBandRecord, params })
 
-    if (!connectionRecord.invitation) {
-      throw new AriesFrameworkError('Connection invitation is missing.')
-    }
+    const { outOfBandMessage } = outOfBandRecord
+    const { goal, goalCode, routing, autoAcceptConnection } = params
+
+    const connectionRecord = await this.connectionService.createConnection({
```

Comment:
Is the idea to extract the connection protocol stuff to a separate `ConnectionProtocol` class?

Reply:
Yes.